### PR TITLE
prepare for release v1.0.1.8

### PR DESCRIPTION
**Pull request by haybankz**
### Descripition

I have 
- [ ] removed "All file" permission for android 11 and above
- [ ] migrated to the published pylons_sdk
- [ ] moved nft.storage api key to a gitignored file

### CODE REVIEW

File: lib/models/storage_response_model.dart [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      this.pin, 
+      // this.deals
+  });
+
+  Value.fromJson(dynamic json) {
+    cid = json['cid'];
+    created = json['created'];
+    type = json['type'];
+    scope = json['scope'];
+    // if (json['files'] != null) {
```
Comment (mustafapylons) [inline lib/models/storage_response_model.dart]: Unless some code is commented for testing purpose, there shouldn't be any dead code like this. Please run a lint utility to ensure clean code.

File: lib/models/storage_response_model.dart [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    //   files = [];
+    //   json['files'].forEach((v) {
+    //     files?.add(dynamic.fromJson(v));
+    //   });
+    // }
+    size = json['size'];
+    pin = json['pin'] != null ? Pin.fromJson(json['pin']) : null;
+    // if (json['deals'] != null) {
```
Comment (mustafapylons) [inline lib/models/storage_response_model.dart]: Same here, dead code to be removed.

File: lib/screens/publish_screen.dart [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+                        Positioned(
+                            top: 60,
+                            right: 10,
+                            child: RoundedPurpleButtonWidget(
+                              onPressed: () {
+                                provider.shareNFT();
+                              },
+                              icon: kShareIcon,
+                            )),
+                        // Positioned(
```
Comment (mustafapylons) [inline lib/screens/publish_screen.dart]: Is this necessary for testing etc, otherwise needs to be removed

File: lib/screens/publish_screen.dart [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+                          ),
+                          Text(
+                            "Royalty: ${provider.royaltyController.text}%",
+                            style:
+                                Theme.of(context).textTheme.caption!.copyWith(
+                                      fontSize: 14,
+                                    ),
+                          ),
+                          // const Divider(
+                          //   height: 40,
```
Comment (mustafapylons) [inline lib/screens/publish_screen.dart]: More lint'ing needs to be done here.

File: lib/models/storage_response_model.dart [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    var map = <String, dynamic>{};
+    map['ok'] = ok;
+    if (value != null) {
+      map['value'] = value?.toJson();
+    }
+    return map;
+  }
+}
+
+/// cid : "bafkreihlusofe353lfwntp7xcwvpzjvqzsohb7iykni3z7525gidsg3wqa"
```
Comment (mustafapylons) [inline lib/models/storage_response_model.dart]: Is this some test data that's commented?

File: lib/models/storage_response_model.dart [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    map['size'] = size;
+    if (pin != null) {
+      map['pin'] = pin?.toJson();
+    }
+    return map;
+  }
+}
+
+/// cid : "bafkreihlusofe353lfwntp7xcwvpzjvqzsohb7iykni3z7525gidsg3wqa"
+/// created : "2021-11-08T13:34:51.769+00:00"
```
Comment (mustafapylons) [inline lib/models/storage_response_model.dart]: same here, more test data.

File: lib/screens/description_screen.dart [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+                          FilteringTextInputFormatter.digitsOnly,
+                          LengthLimitingTextInputFormatter(5),
+                          AmountFormatter(maxDigits: 5)
+                        ],
+                        controller: provider.noOfEditionController,
+                        validator: (value) {
+                          if (value!.isEmpty) return "Enter number of editions";
+                          if (int.parse(value.replaceAll(",", "")) < kMinValue)
+                            return "Minimum is $kMinValue";
+                          if (int.parse(value.replaceAll(",", "")) >
```
Comment (mustafapylons) [inline lib/screens/description_screen.dart:153]: better formatting here, please lint your code for readability

File: lib/widgets/rounded_purple_button_widget.dart [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+          gradient: const LinearGradient(
+            begin: Alignment.topLeft,
+            end: Alignment.bottomRight,
+            stops: [0.0, 1.0],
+            colors: [
+              EaselAppTheme.kWhite02,
+              EaselAppTheme.kPurple06,
+            ],
+          ),
+          // color: Color(0xFF1212C4),
```
Comment (mustafapylons) [inline lib/widgets/rounded_purple_button_widget.dart]: what's this for?

**Review by mustafapylons — COMMENTED** [review]
I think these issues will disappear after we activate our lint.

File: lib/models/storage_response_model.dart (hunk 5 above)
Comment (haybankz) [reply lib/models/storage_response_model.dart]: This model class is used to parse an api endpoint json data. The comment is there to help understand how the response json data looks like.

File: lib/models/storage_response_model.dart (hunk 6 above)
Comment (haybankz) [reply lib/models/storage_response_model.dart]: Same here.

File: lib/models/storage_response_model.dart (hunk 5 above)
Comment (mustafapylons) [reply lib/models/storage_response_model.dart]: These kind of json data can be in a file, in a design document. It has no place in the actual code as a comment. 
Commented code shouldn't be in the PRs. Comments should help:
1- Make the code more readable
2- Help build out documentation.

**Review by mustafapylons — COMMENTED** [review]
With linting looks better. Remove any other commented code and we are good to go for approval.

**Comment by haybankz** [comment]
@mustafapylons  commented code has been removed, so i'm going to merge because Corey and I need the code

**Pull request merged**